music21/roman.py

```diff
@@ -628,105 +628,47 @@ def correctRNAlterationForMinor(figureTuple, keyObj):
-    alternatively as #vi and #vii:
-
-    >>> romanNumeral3 = roman.romanNumeralFromChord(
-    ...     chord.Chord(['A4', 'C5', 'E-5']),
-    ...     key.Key('c'),
-    ...     )
-    >>> romanNumeral3
-    <music21.roman.RomanNumeral vio in c minor>
-
-    >>> romanNumeral4 = roman.romanNumeralFromChord(
-    ...     chord.Chord(['A-4', 'C5', 'E-5']),
-    ...     key.Key('c'),
-    ...     )
-    >>> romanNumeral4
-    <music21.roman.RomanNumeral bVI in c minor>
-
-    >>> romanNumeral5 = roman.romanNumeralFromChord(
-    ...     chord.Chord(['B4', 'D5', 'F5']),
-    ...     key.Key('c'),
-    ...     )
-    >>> romanNumeral5
-    <music21.roman.RomanNumeral viio in c minor>
-
-    >>> romanNumeral6 = roman.romanNumeralFromChord(
-    ...     chord.Chord(['B-4', 'D5', 'F5']),
+    >>> roman.romanNumeralFromChord(
+    ...     chord.Chord(['C4', 'E-4', 'G4']),
     ...     key.Key('c'),
     ...     )
-    >>> romanNumeral6
-    <music21.roman.RomanNumeral bVII in c minor>
-
-    Diminished and half-diminished seventh chords can omit the third and still
-    be diminished: (n.b. we also demonstrate that chords can be created from a
-    string):
+    <music21.roman.RomanNumeral i in c minor>
 
-    >>> romanNumeralDim7 = roman.romanNumeralFromChord(
-    ...     chord.Chord('A3 E-4 G-4'),
-    ...     key.Key('b-'),
-    ...     )
-    >>> romanNumeralDim7
-    <music21.roman.RomanNumeral viio7 in b- minor>
-
-    For reference, odder notes:
-
-    >>> romanNumeral7 = roman.romanNumeralFromChord(
-    ...     chord.Chord(['A--4', 'C-5', 'E--5']),
-    ...     key.Key('c'),
+    >>> roman.romanNumeralFromChord(
+    ...     chord.Chord(['E-3', 'C4', 'G-6']),
+    ...     key.Key('g#'),
     ...     )
-    >>> romanNumeral7
-    <music21.roman.RomanNumeral bbVI in c minor>
+    <music21.roman.RomanNumeral bivo6 in g# minor>
 
-    >>> romanNumeral8 = roman.romanNumeralFromChord(
-    ...     chord.Chord(['A#4', 'C#5', 'E#5']),
-    ...     key.Key('c'),
-    ...     )
-    >>> romanNumeral8
-    <music21.roman.RomanNumeral #vi in c minor>
+    If no such keyObj is provided, this function will still run with
+    its own choice of a plausible key.
+    In most contexts, the guessed key corresponds to the chord:
+    the key's tonic is given by the chord's root, and
+    the key's quality (major or minor) comes from the (major or minor)
+    third between the chord's root and its third.
 
-    >>> romanNumeral10 = roman.romanNumeralFromChord(
-    ...     chord.Chord(['F#3', 'A3', 'E4', 'C5']),
-    ...     key.Key('d'),
-    ...     )
-    >>> romanNumeral10
-    <music21.roman.RomanNumeral #iiiø7 in d minor>
+    >>> roman.romanNumeralFromChord(
+    ...     chord.Chord(['C4', 'E-4', 'G4']),
+    ...     )  # NB: no key
+    <music21.roman.RomanNumeral i in c minor>
 
+    >>> roman.romanNumeralFromChord(
+    ...     chord.Chord(['C4', 'E4', 'G4']),
+    ...     )  # NB: no key and nor major third
```

Comment:
I don't understand this comment.

Reply:
Me neither

Reply:
Also, better to define the chord in the line before as a variable (cMaj) and then pass it to roman.romanNumeralFromChord(cMaj) .  Inside doctests, comments shouldn't be necessary, since they can be put as descriptions of the test before.